pyop2/sparsity.pyx

```diff
@@ -141,17 +141,16 @@ def build_sparsity(sparsity):
                           bsize=1)
     preallocator.setUp()
 
-    iteration_regions = sparsity.iteration_regions
     if mixed:
         for i, r in enumerate(rset):
             for j, c in enumerate(cset):
-                maps = list(zip((m.split[i] for m in sparsity.rmaps),
-                                (m.split[j] for m in sparsity.cmaps)))
+                maps = sparsity.rcmaps[(i, j)] if (i, j) in sparsity._maps_and_regions else []
+                iter_regions = sparsity.iteration_regions[(i, j)] if (i, j) in sparsity._maps_and_regions else []
```

Comment:
```suggestion
                if (i, j) in sparsity._map_and_regions:
                    maps = sparsity.rcmaps[(i, j)]
                    iter_regions = sparsity.iteration_regions[(i, j)]
```
or something. I don't think it's nice to have the same ternary operator duplicated on adjacent lines.

Reply:
Or alternatively just use `sparsity.rcmaps.get((i, j), [])`.

Reply:
Indeed.